engine/src/retrier.rs

```diff
@@ -400,11 +400,17 @@ where
 							let half_max = max_sleep_duration(initial_request_timeout, attempt) / 2;
 							let sleep_duration = half_max + rand::thread_rng().gen_range(Duration::default()..half_max);
 
-							let error_message = format!("Retrier {name}: Error for request `{request_log}` with id `{request_id}`, attempt `{attempt}`: {e}. Delaying for {:?}", sleep_duration);
-							if attempt == 0 && !matches!(retry_limit, RetryLimit::Limit(1)) {
-								tracing::warn!(error_message);
-							} else {
-								tracing::error!(error_message);
+							let attempt_number = attempt.saturating_add(1);
+							match retry_limit {
+								RetryLimit::Limit(max_attempts) if attempt_number >= max_attempts => {
```

Comment:
In the past we would also log error (rather than a warn) in the specific case of `Limit = 1`. Looks like you code changes that. I'm assuming it is not intentional?

Reply:
It was intentional. The old logic did not make sense to me, why a limit of 1? It was not clear why or when it would log as warn and when it would log as error. So now it is clear, only log error on the last try. Makes more sense.

Reply:
As discussed, the error is still printed in case `Limit = 1` as it would hit the first branch.

Reply:
@kylezs Maybe you can clear up the intent behind when we want this to be logged as error/warn?
1) Log error on last try.
2) Log error on last try and any try after first.

Reply:
The real solution to this is this: https://linear.app/chainflip/issue/PRO-1206/timeouts-per-rpc-retrier-query-type so that we can just have them all be errors but adjust the times based on which we expect to take longer / shorter times.

> 1.  Log error on last try.
> 2. Log error on last try and any try after first.

By 2. do you mean just "any try after first" which would include the last try? This is how the code works at the moment, and is the intention. So I think we want to just leave this as is. Feel free to leave a comment there though, as I can see how that might not be clear.

Reply:
> do you mean just "any try after first" which would include the last try

No, If there is no retries, then the last try is also the first, so it would not be an error. I think it's clearer to say last try and any try after first.

Reply:
Ok, but the code still works like before this change anyway? The current change just splits 2 conditions into 3.

Reply:
If it's the first attempt and there are many retries (with limit or without limit) then warn.
Otherwise log error for every attempt. (This condition includes:  If it's the first attempt and there's only one try then it will error.)